### PR TITLE
EN-6404 : refactor size check unmarshalizer

### PR DESCRIPTION
**Pull request by miiu96**
Refactored the size check unmarshalizer in order to optimize how it works
Instead of doing a marshal operation on the received obj will try to do a type assertion to convert obj (type interface{} ) in a Sizer type obj . If type assertion works fine new object of type `Sizer` can return size in bytes of the received object.

### CODE REVIEW

File: marshal/testSizeCheckUnmarshal/proto/testStruct.proto [hunk 1]
```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+package testSizeCheckUnmarshal;
+
+option (gogoproto.stable_marshaler_all) = true;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+message TestStruct1  {
+uint64                   field1               = 1;
```
Comment (AdoAdoAdo) [inline marshal/testSizeCheckUnmarshal/proto/testStruct.proto:10]: check the indentation

File: marshal/testSizeCheckUnmarshal/testStructs_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+	ts1 := &TestStruct1{
+		Field1: 1,
+		Field2: []byte("filed2"),
+	}
+	ts2 := &TestStruct2{}
+
+	marshalizedTs1, _ := sizeCheckMarslalizer.Marshal(ts1)
+	err := sizeCheckMarslalizer.Unmarshal(ts2, marshalizedTs1)
+	fmt.Println(ts2)
```
Comment (AdoAdoAdo) [inline marshal/testSizeCheckUnmarshal/testStructs_test.go:62]: you can remove the print

File: marshal/testSizeCheckUnmarshal/testStructs_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	ts2 := &TestStruct2{
+		Field1: 1,
+		Field2: []byte("field2"),
+		Field3: []byte("field3"),
+	}
+	ts1 := &TestStruct1{}
+
+	marshalizedTs2, _ := sizeCheckMarslalizer.Marshal(ts2)
+	err := sizeCheckMarslalizer.Unmarshal(ts1, marshalizedTs2)
+	assert.NotNil(t, err)
```
Comment (AdoAdoAdo) [inline marshal/testSizeCheckUnmarshal/testStructs_test.go:45]: can you verify the error is ErrUnmarshallingBadSize instead?

File: marshal/testSizeCheckUnmarshal/proto/testStruct.proto (hunk 1 above)
Comment (miiu96) [reply marshal/testSizeCheckUnmarshal/proto/testStruct.proto:10]: done.

File: marshal/testSizeCheckUnmarshal/testStructs_test.go (hunk 2 above)
Comment (miiu96) [reply marshal/testSizeCheckUnmarshal/testStructs_test.go:62]: done.

File: marshal/testSizeCheckUnmarshal/testStructs_test.go (hunk 3 above)
Comment (miiu96) [reply marshal/testSizeCheckUnmarshal/testStructs_test.go:45]: done.

**Pull request merged**